### ISSUE TITLE
functorch import messed up python logging module

### ISSUE THREAD

**Issue by xwang233** [pytorch/pytorch#85952]
### 🐛 Describe the bug

Run following script:

```python
import logging

# from functorch.compile import memory_efficient_fusion

_logger = logging.getLogger('aaaaa')

logging.root.addHandler(logging.StreamHandler())
logging.root.setLevel(logging.INFO)
    
_logger.info('some log info')
```

Without functorch import line:
```
some log info
```

With functorch import line:
```
INFO:aaaaa:some log info
some log info
```

I think import functorch shouldn't mess up the global logging handler state?

### Versions

This behavior was first seen on 9/27/22 and the log was working normally 9/26/22, so it probably happened between the below package updates:

Python version: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-80-generic-x86_64-with-glibc2.17

9/27
torch_commit: https://github.com/pytorch/pytorch/commit/f64857189d514a662dac09ca6421e1e95e87e843
torchdynamo_commit: https://github.com/pytorch/torchdynamo/commit/6ead5cae0d1234aa64db06fe230ef56e12ec76fe
torchvision_commit: https://github.com/pytorch/vision/commit/a35be97a6e6725c83032315a8f5e5f6911c9ef41

9/26
torch_commit: https://github.com/pytorch/pytorch/commit/9c036aa112b0a8fd9afb824d1fda058e2b66ba1d
torchdynamo_commit: https://github.com/pytorch/torchdynamo/commit/41c44bc1d080d6cf063419a4166732b983b84eef
torchvision_commit: https://github.com/pytorch/vision/commit/7046e56fe4370e94339b3e8b6fd011e285294a3a

cc @ezyang @gchanan @zou3519 @SherlockNoMad @soumith @Chillee @samdow @ptrblck

**Comment by zou3519** [comment]
Yes, imports should not mess up global state. We will take a look

**Comment by ezyang** [comment]
cc @jansel @SherlockNoMad

**Comment by zou3519** [comment]
Bisected to https://github.com/pytorch/pytorch/pull/85617 . It should not be unconditionally adding defaults to the global logger: https://github.com/pytorch/pytorch/blob/d28a882319d92bae17827101787adf838a05df0a/torch/fx/passes/utils/matcher_utils.py#L14-L17

cc @SherlockNoMad how do you want to proceed? A simple fix is to delete those two lines, but I don't have context around htis code.

**Comment by SherlockNoMad** [comment]
> Bisected to #85617 . It should not be unconditionally adding defaults to the global logger:
> 
> https://github.com/pytorch/pytorch/blob/d28a882319d92bae17827101787adf838a05df0a/torch/fx/passes/utils/matcher_utils.py#L14-L17
> 
> cc @SherlockNoMad how do you want to proceed? A simple fix is to delete those two lines, but I don't have context around htis code.

Please feel free to delete these two line to unblock you.

**Comment by xwang233** [comment]
Hi @zou3519 , I think this issue still exists in the latest master. See https://github.com/pytorch/pytorch/blob/master/torch/fx/passes/utils/matcher_utils.py#L14-L29

**Comment by xwang233** [comment]
Probably modified by this PR https://github.com/pytorch/pytorch/pull/86421 cc @SherlockNoMad

**Comment by malfet** [comment]
@xwang233 do you mind creating separate issue rather than reopening this one? As this one reported a functorch regression, which were fixed already, isn't it?

**Comment by xwang233** [comment]
It still exists in the current master.

**Comment by SherlockNoMad** [comment]
I tried to reproduce locally. Reverting my change doesn't resolve the problem. 

Also, I noticed that `import functorch` doesn't trigger the problem. `import torchdynamo` tiggers the problem.

**Comment by SherlockNoMad** [comment]
Actually, I found that adding `logging.basicConfig()` would trigger the problem. 

I removed the line `logging.basicConfig(level=logging.WARNING)` in partitioner.py, and the problem is then resolved. 

I will have a PR to fix this...
